### PR TITLE
block_with_iommu: New case to reload kernel with iommu

### PR DESCRIPTION
**Pull request by YongxueHong**
New a case to use command kexec to reload kernel under
the IOMMU mode.

ID: 1822052
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
hi @qingwangrh  @zhenyzha   pls help me to review, thanks.

File: qemu/tests/block_with_iommu.py [hunk 1]
```diff
@@ -13,12 +14,32 @@ def run(test, params, env):
         1. Add "intel_iommu=on" to kernel line of q35 guest.
```
Comment (zhenyzha) [inline qemu/tests/block_with_iommu.py:14]: This is x86 only case, please ignore me.

Comment (YongxueHong) [reply qemu/tests/block_with_iommu.py:14]: Sorry for inconvenience.

**Comment by qingwangrh** [comment]
python ConfigTest.py  --testcase=block_with_iommu --iothread_scheme=roundrobin --nr_iothreads=2 --platform=x86_64 --guestname=RHEL.8.3.0 --driveformat=virtio_scsi --nicmodel=virtio_net --imageformat=qcow2 --machines=q35 --customsparams="image_aio=native\nqemu_force_use_drive_expression = no\ncd_format=ide" --clone=yes

(1/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1063.81 s)
 (2/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.rh_kernel_update.q35: PASS (245.79 s)
 (3/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.block_with_iommu.verify_enabled.q35: PASS (59.76 s)
 (4/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.block_with_iommu.verify_enabled.reload_kernel.q35: PASS (62.88 s)

**Pull request merged**